mcstatus/protocol/connection.py

```diff
@@ -1,339 +1,596 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"TCP and UDP Connections, both asynchronous and not."
+
 from __future__ import annotations
 
 import asyncio
 import socket
 import struct
 from abc import ABC, abstractmethod
-from ctypes import c_int32 as signed_int32
-from ctypes import c_uint32 as unsigned_int32
 from ipaddress import ip_address
-from typing import Iterable, Optional, SupportsBytes, TYPE_CHECKING, Union
+from typing import Iterable, Optional, Union
 
-import asyncio_dgram
-
-from mcstatus.address import Address
+from ctypes import c_uint32 as unsigned_int32
+from ctypes import c_int32 as signed_int32
+from ctypes import c_uint64 as unsigned_int64
+from ctypes import c_int64 as signed_int64
 
-if TYPE_CHECKING:
-    from typing_extensions import SupportsIndex  # Python 3.7 doesn't support this yet.
+from typing_extensions import SupportsIndex
```

Comment:
```suggestion
if TYPE_CHECKING:
    from typing_extensions import SupportsIndex  # Python 3.7 doesn't support this yet.

    BytesConvertable = Union[SupportsIndex, Iterable[SupportsIndex]]
```
The conditional import is there so this does not cause an ImportError at runtime. mcstatus does not list `typing-extensions` as a runtime dependency.

Reply:
That makes sense, but I'm confused how BytesConvertable is able to be used even though it shouldn't exist if TYPE_CHECKING is false

Reply:
We're using `from futures import __annotations__` which automatically treats every annotation as a string which means it won't cause a runtime error